std/validation/assert.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+export class AssertionFailedError extends Error {
```

Comment:
Introducing a superficial difference to avoid a name clash is especially confusing. That's why I suggested `StdInternalError`... of course that name only works if it's being used for debugging and not for user input validation somewhere 😫